spyder/widgets/variableexplorer/utils.py

```diff
@@ -174,6 +174,14 @@ def datestr_to_datetime(value):
     return v
 
 
+def str_to_timedelta(value):
+    m = re.match('^(?:datetime\.timedelta)?\(?([^\)]*)\)?$', value)
+    if not m:
+        raise ValueError('Invalid string for datetime.timedelta')

```

Comment:
I don't think we should return an Error because Spyder can freeze because of it.

I think it's better to return the original value. That way people would understand they did something wrong when writing a new timedelta.

Reply:
Please solve this one, and I'll merge.